### PR TITLE
Use adwaita in the prefs dialog

### PR DESCRIPTION
**Pull request by philippun1**
This PR will use the Adwaita preferences classes for the prefs dialog and sub dialogs:

- [x] Split the giant Settings.ui in separate .ui files for each page/dialog (which makes it much more readable)
- [x] Use AdwPreferncePage for the tabs
- [x] Use AdwPreferenceGroup and AdwPreferenceRow for the settings (pages + dialogs)

**Optional:**
- [x] Find more suitable icons for the preference pages
- Group titles in sub dialogs

**Backwards compatibility:**
This PR is not backwards compatible to Gnome 41.

### CODE REVIEW

**Comment by charlesg99** [comment]
Thanks Philipp! I feel dumb to ask, but how do you access this new settings UI? :joy:

Also, I guess that since the prefs.js file is changing, this will only be compatible with gnome-shell >= 42, correct?

**Comment by philippun1** [comment]
Gnome 42 will use _fillPreferencesWindow_ instead of _buildPrefsWidget_ if avialable. So if you try the branch on Gnome 42 it will automatically use the new ui. The pages itself are migrated now, but not functional yet.

I will implement a switch if _fillPreferencesWindow_ or _buildPrefsWidget_ is being used to load the old ui or the new ui into the builder. Best case we can stay compatible to Gnome 41. But the only distribution I know of which uses it is Fedora 35, which is EOL by the end of the year. So we can remove 41 from the compatible list, otherwise we would have to maintain two ui files.

**Comment by charlesg99** [comment]
Got it, thanks!

**Comment by philippun1** [comment]
@charlesg99 This is almost done now, I've updated the description. The only missing thing is the GtkListBox style (background color mismatch).
I've tested every option (but the import/export). But it would be good if someone else could test it.
@crab2313 @fcole90 @miigotu maybe you are interested.

**Comment by philippun1** [comment]
Fixed the background color mismatch of the GtkListBox taskbar_display_listbox.

File: ui/SettingsPosition.ui [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+  <object class="GtkAdjustment" id="panel_length_adjustment">
+    <property name="upper">100</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+  </object>
+
+  <object class="AdwPreferencesPage" id="position">
+    <property name="title">Position</property>
+    <property name="icon_name">find-location</property>
```
Comment (AndrewZaech) [inline ui/SettingsPosition.ui:20]:
```suggestion
    <property name="icon_name">find-location-symbolic</property>
```
Allows for icon color adjusting with system theme.

File: ui/SettingsPosition.ui [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        <object class="AdwPreferencesGroup" id="position_group_panel">
+          <property name="title" translatable="yes">Panel</property>
+
+          <child>
+            <object class="AdwActionRow">
+              <property name="title" translatable="yes">Display the main panel on</property>
+              <child>
+                <object class="GtkSwitch" id="multimon_multi_switch">
+                  <property name="valign">center</property>
+                </object>
```
Comment (AndrewZaech) [inline ui/SettingsPosition.ui:33]:
```suggestion
                <object class="GtkComboBoxText" id="multimon_primary_combo">
                  <property name="can_focus">False</property>
                  <property name="valign">center</property>
                </object>
```
This is mixed up with "Display panels on all monitors"

File: ui/SettingsPosition.ui [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+          </child>
+
+          <child>
+            <object class="AdwActionRow">
+              <property name="title" translatable="yes">Display panels on all monitors</property>
+              <child>
+                <object class="GtkComboBoxText" id="multimon_primary_combo">
+                  <property name="can_focus">False</property>
+                  <property name="valign">center</property>
+                </object>
```
Comment (AndrewZaech) [inline ui/SettingsPosition.ui:45]:
```suggestion
                <object class="GtkSwitch" id="multimon_multi_switch">
                  <property name="valign">center</property>
                </object>
```

File: ui/SettingsAbout.ui [hunk 4]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<interface>
+  <requires lib="gtk" version="4.0"/>
+
+  <object class="AdwPreferencesPage" id="about">
+    <property name="title">About</property>
+    <property name="icon_name">help-about</property>
```
Comment (AndrewZaech) [inline ui/SettingsAbout.ui:7]:
```suggestion
    <property name="icon_name">help-about-symbolic</property>
```

File: ui/SettingsAction.ui [hunk 5]
```diff
@@ -0,0 +1,172 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<interface>
+  <requires lib="gtk" version="4.0"/>
+
+  <object class="AdwPreferencesPage" id="action">
+    <property name="title">Action</property>
+    <property name="icon_name">view-pin</property>
```
Comment (AndrewZaech) [inline ui/SettingsAction.ui:7]:
```suggestion
    <property name="icon_name">view-pin-symbolic</property>
```

File: ui/SettingsBehavior.ui [hunk 6]
```diff
@@ -0,0 +1,191 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<interface>
+  <requires lib="gtk" version="4.0"/>
+  <object class="AdwPreferencesPage" id="behavior">
+    <property name="title">Behavior</property>
+    <property name="icon_name">visible</property>
```
Comment (AndrewZaech) [inline ui/SettingsBehavior.ui:6]:
```suggestion
    <property name="icon_name">preferences-system-symbolic</property>
```

File: ui/SettingsFineTune.ui [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+  <object class="GtkAdjustment" id="leftbox_padding_adjustment">
+    <property name="lower">0.33</property>
+    <property name="upper">1</property>
+    <property name="step_increment">0.01</property>
+    <property name="page_increment">0.1</property>
+  </object>
+
+  <object class="AdwPreferencesPage" id="finetune">
+    <property name="title">Fine-Tune</property>
+    <property name="icon_name">color-select</property>
```
Comment (AndrewZaech) [inline ui/SettingsFineTune.ui:42]:
```suggestion
    <property name="icon_name">preferences-other-symbolic</property>
```

File: ui/SettingsStyle.ui [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+
+  <object class="GtkAdjustment" id="trans_gradient_opacity2_adjustment">
+    <property name="upper">100</property>
+    <property name="step_increment">5</property>
+    <property name="page_increment">10</property>
+  </object>
+
+  <object class="AdwPreferencesPage" id="style">
+    <property name="title">Style</property>
+    <property name="icon_name">image-adjust-color</property>
```
Comment (AndrewZaech) [inline ui/SettingsStyle.ui:39]:
```suggestion
    <property name="icon_name">applications-graphics-symbolic</property>
```

**Comment by philippun1** [comment]
@AndrewZaech Thanks for the review, I applied your suggestions.
@charlesg99 Is there anything missing before we can merge this? In my opinion it would be ready now.

**Comment by charlesg99** [comment]
Thanks Philipp! I'm thinking about releasing another version early next week and it should be the last one supporting GNOME 41, so I guess we can merge this right after and move forward with 42. I still need to test this more but don't worry, now that GNOME properly match extensions/gnome-shell by versions and makes the update process hassle free, I'm planning on releasing more often.

**Comment by philippun1** [comment]
Great! I removed all the backwards compatibility code in prefs.js. The only thing missing is the Settings.ui because it would result in conflicts with master branch.

**Comment by philippun1** [comment]
Rebased and added the missing commits from master.

**Comment by philippun1** [comment]
@charlesg99 Any news on this one? Would be cool to have an adwaita preference dialog with Gnome 43. Full immersion :rocket:

**Comment by charlesg99** [comment]
Hey Philipp, yes sorry I haven't gotten to testing it yet. I'll try to include it in the next release, as we're getting closer to gnome 41 EOL.

**Comment by charlesg99** [comment]
Hey Philipp, sorry it took so long! I'm planning another gnome 43 maintenance release before starting changes (if any) for gnome 44. I'd like to merge this PR so we can test, is it still ok? I don't think we added any more settings in the dialog so *I think* this should work fine. Thanks!

**Comment by philippun1** [comment]
@charlesg99 Sure, you can merge it.

**Comment by philippun1** [comment]
Sorry, there was one commit left. I cherry picked it and updated the corresponding SettingsAbout.ui.
But now everything is ready for merge.

**Comment by charlesg99** [comment]
Thanks Philipp! I did a first round of tests this morning and everything seems to work!

**Comment by charlesg99** [comment]
Thanks Philipp! I'll leave it on master for a week or so and then make a release.

**Pull request merged**